pkg/cache/cache_test.go

```diff
@@ -42,70 +41,70 @@ func (m *TestMemcache) Set(i *memcache.Item) error {
 	return nil
 }
 
-func TestReplicatedMemcacheWithPartialTimeout(t *testing.T) {
-	m := ReplicatedMemcached{
-		prefix:    "test",
-		timeoutMs: 20,
-		instances: []Cache{
-			&TestMemcache{
-				delayMs: 1,
-			},
-			&TestMemcache{
-				delayMs: 10,
-			},
-			&TestMemcache{
-				// this one will timeout every time
-				delayMs: 100,
-			},
-		},
-	}
-
-	aData := []byte("aval")
-	bData := []byte("bval")
-	m.Set("a", aData, 0)
-	m.Set("b", bData, 0)
-
-	aRes, err := m.Get("a")
-	if !cmp.Equal(aRes, aData) {
-		t.Fatalf("Expected %v value for key %s, got %v", aData, "a", aRes)
-	}
-	if err != nil {
-		t.Fatalf("Error while getting value for key %s; %v", "a", err)
-	}
-
-	xRes, err := m.Get("x")
-	if err != ErrNotFound {
-		t.Fatalf("Expected cache miss, that did not happen. Got %v and err %v instead", xRes, err)
-	}
-}
-
-func TestReplicatedMemcacheTimeout(t *testing.T) {
-	m := ReplicatedMemcached{
-		prefix:    "test",
-		timeoutMs: 10,
-		instances: []Cache{
-			&TestMemcache{
-				delayMs: 250,
-			},
-			&TestMemcache{
-				delayMs: 230,
-			},
-			&TestMemcache{
-				delayMs: 100,
-			},
-		},
-	}
-
-	aData := []byte("aval")
-	bData := []byte("bval")
-	m.Set("a", aData, 0)
-	m.Set("b", bData, 0)
-
-	aRes, err := m.Get("a")
-	if err == nil || err == ErrNotFound {
-		t.Fatalf("Expected timeout, got val %v, err %v", aRes, err)
-	}
-}
+// func TestReplicatedMemcacheWithPartialTimeout(t *testing.T) {
```

Comment:
Out of curiosity, why the tests got commented out?

Reply:
Because I have removed redundant polymorphism. We cannot have this test without it.

Reply:
@timtofan This is the sacrifice we will have to make to make the code more clear and get access to other memcache ops moving forward. 
In general, in my experience, it is usually not a great idea to unit-test things that require remote connections. We can do it, but the overhead is not worth it.

Reply:
@timtofan We have also discussed this with @emadolsky and agreed not to do unit testing when it has high overhead. In my opinion, this is one such case.

Reply:
I think, it makes more sense to remove the tests, not comment them out. So, I did that.

Reply:
thanks for the info